word_racer/session.py

```diff
@@ -0,0 +1,295 @@
+
+_FREQ = [[142,25,66,53,218,15,42,34,168,1,8,88,44,130,124,45,1,135,180,125,50,10,7,2,23,4],
+         [130,27,65,53,199,17,42,35,154,2,10,86,44,118,114,45,2,123,164,115,50,12,9,3,24,6],
+         [119,29,64,53,181,19,43,37,142,3,13,84,45,108,104,46,3,112,150,105,51,15,12,5,26,8],
+         [100,33,62,53,150,24,44,40,120,9,21,80,46,90,87,47,9,93,125,88,52,23,20,12,30,16]]
+
+_BONUSES = [{}, {(0,2):2,(5,3):2}, {(0,0):3,(5,5):3}, {(0,0):3,(0,5):2,(5,0):2,(5,5):3}]
+
+class WordRacerSession:
+    def __init__(self, ctx):
+        self.level = 0
+        self.dataDir = pathlib.Path(__file__).parent.resolve() / 'data'
+        
+        # feel free to experiment with this
+        self.dictDir = self.dataDir/"dict/enable2k.txt"
+        self.fontDir = self.dataDir/"fonts/Roboto-Medium.ttf"
+
+        self.ctx = ctx
+        self.output_image_path = self.dataDir / f'board-{ctx.channel.id}.png'
+        self.scores = Counter()
+        self.round_scores = Counter()
+        self.claims = {}
+        self.valid_words = Counter()
+        self.board = [["_" for _ in range(6)] for __ in range(6)]
+        self.bonus = {}
+        self.nrows = 6
+        self.ncols = 6
+        self.round_finish = False
+        self.reaction_queue = []
+
+    @classmethod
+    def start(cls, ctx):
+        session = cls(ctx)
+        loop = ctx.bot.loop
+        session._task = loop.create_task(session.run())
+        return session
+
+    async def run(self):
+        
+        await self._send_startup_msg()
+
+        # Round loop
+        while self.level < _LEVEL_COUNT:
+            # Round setup
+            self.claims = {}
+            self.valid_words = Counter()
+            self.round_scores = Counter()
+            self._gen_board()
+            self._get_score_dict()
+            self._gen_image()
+            await asyncio.sleep(3)
+
+            # send board image
+            f = discord.File(str(self.output_image_path))
+            await self.ctx.send(f"Starting round {self.level+1}. {len(self.valid_words)} words to find.",file=f)
+
+            # Message handler for round
+            self.round_finish = False
+            await self.run_round()
+
+            # Round cleanup
+            await self.finish_round()
+            self.level += 1
+            
+        await self.end_game()
+
+    async def _send_startup_msg(self):
+        await self.ctx.send("Starting Word Racer. Find words boggle-style and gain points."
+                            " Incorrect calls are -1 point. In rounds 2-4 there are bonuses:"
+                            " blue is 2x points and red is 3x points (they multiplicatively stack)."
+                            " Good luck!")
+
+    async def run_round(self):
+        timer_task = self.ctx.bot.loop.create_task(self.timer_task())
+        reaction_task = self.ctx.bot.loop.create_task(self.reactions_handler())
+        try:
+            await self.ctx.bot.wait_for("message", check=self.check_message, timeout=120)
```

Comment:
It would be nice to have a global constant to handle the timeout value here and how it relates to the value hard-coded in timer_task (the former value being a likely target for a config option later).